src/machine/machined-varlink.c

```diff
@@ -489,6 +491,50 @@ static int vl_method_get(sd_varlink *link, sd_json_variant *parameters, sd_varli
         return lookup_machine_by_name(link, m, mn);
 }
 
+static int vl_method_get_by_pid(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        Manager *m = ASSERT_PTR(userdata);
+
+        int r;
+        pid_t pid = 0;
+        assert_cc(sizeof(pid_t) == sizeof(uint32_t));
+        assert(parameters);
+
+        const sd_json_dispatch_field dispatch_table[] = {
+                { "pid", SD_JSON_VARIANT_UNSIGNED, sd_json_dispatch_uint32, PTR_TO_SIZE(&pid), SD_JSON_MANDATORY },
+                {}
+        };
+
+        r = sd_varlink_dispatch(link, parameters, dispatch_table, 0);
```

Comment:
The same. Please make the table static, and pass `&pid` to `sd_varlink_dispatch()`.